Sources/Danger/CliArgs.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  CliArgs.swift
+//  Danger
+//
+//  Created by Franco Meloni on 12/11/2018.
+//
+
+// Describes the possible arguments that could be used when calling the CLI
+public struct CliArgs {
+    // So you can have many danger runs in one code review
+    public let id: String?
+    
+    // The base reference used by danger PR (e.g. not master)
+    public let base: String?
+    
+    // For debugging
+    public let verbose: String?
+    
+    // Used by danger-js o allow having a custom CI
+    public let externalCiProvider: String?
+    
+    // textOnly
+    public let textOnly: String?
+    
+    // A custom path for the dangerfile (can also be a remote reference)
+    public let dangerfile: String?
+    
+    fileprivate enum CodingKeys: String {
+        case id
+        case base
+        case verbose
+        case externalCiProvider
+        case textOnly
+        case dangerfile
+    }
+    
+    init(dictionary: [String:Any]) {
```

Comment:
I didn't want to make this part of the parsed Danger struct - or of the Danger runner (because on the `Dangerfile` you probably don't need this, and in the runner you don't need all the other `Danger` related data).
Then didn't make a lot of sense to me to parse the entire DSL with `JSONEncoder`, i preferred to use this approach.
Also `JSONSerialization` is more performant than `Codable`, and given i will have to parse JSON two times (one on the Dangerfile and one on the Runner), i preferred to use `JSONSerialization` than `Codable` in this case

Reply:
This makes sense to me 👍

Reply:
and also protects against it changing and adding new attributes in the future